addons/website_event_track/controllers/event_track.py

```diff
@@ -392,30 +392,64 @@ def event_track_proposal(self, event, **post):
+            if valid_contact_email or post.get('contact_phone'):
+                if visitor_partner and valid_contact_email == visitor_partner.email_normalized:
+                    contact = visitor_partner
+                else:
+                    contact = request.env['res.partner'].sudo().create({
+                        'email': valid_contact_email,
+                        'name': post.get('contact_name'),
+                        'phone': post.get('contact_phone'),
+                    })
+            else:
+                raise exceptions.ValidationError(_("Format Error : please enter a valid contact phone or contact email."))
+        # If the speaker email is the same as logged user's, then also uses its partner on track, same as above.
+        else:
+            valid_speaker_email = tools.email_normalize(post['partner_email'])
+            if visitor_partner and valid_speaker_email == visitor_partner.email_normalized:
+                contact = visitor_partner
+
+        track = request.env['event.track'].with_context({'mail_create_nosubscribe': True}).sudo().create({
             'name': post['track_name'],
+            'partner_id': contact.id,
             'partner_name': post['partner_name'],
-            'partner_email': post['email_from'],
-            'partner_phone': post['phone'],
-            'partner_biography': plaintext2html(post['biography']),
+            'partner_email': post['partner_email'],
+            'partner_phone': post['partner_phone'],
+            'partner_function': post['partner_function'],
+            'contact_phone': contact.phone,
+            'contact_email': contact.email,
             'event_id': event.id,
-            'tag_ids': [(6, 0, tags)],
-            'user_id': False,
+            'tag_ids': [(6, 0, valid_tag_indices)],
             'description': plaintext2html(post['description']),
-            'image': base64.b64encode(post['image'].read()) if post.get('image') else False
+            'user_id': False,
+            'image': base64.b64encode(post['image'].read()) if post.get('image') else False,
         })
+
         if request.env.user != request.website.user_id:
             track.sudo().message_subscribe(partner_ids=request.env.user.partner_id.ids)
-        else:
-            partner = request.env['res.partner'].sudo().search([('email', '=', post['email_from'])])
-            if partner:
-                track.sudo().message_subscribe(partner_ids=partner.ids)
-        return request.render("website_event_track.event_track_proposal", {'track': track, 'event': event})
+
+        return request.render("website_event_track.event_track_proposal", {'track': track, 'event': event, 'main_object': event})
+
+    # ACL : This route is necessary since rpc search_read method in js is not accessible to all users (e.g. public user).
+    @http.route(['''/event/track_proposal/fetch_tags'''], type='json', auth="public", website=True)
```

Comment:
I would propose a naming more orm-based, like
```
/event/track_tag/search_read
```
That we we have a clear mapping between models and methods we try to reach.